doc/userguide/rule-management/rule-reload.rst

```diff
@@ -1,32 +1,70 @@
 Rule Reloads
 ============
 
-Suricata can reload the rules without restarting. This way, there
-is minimal service disruption.
+Suricata was designed to reload rules while it is actively processing
+network traffic to minimize service disruption.
 
-This works by sending Suricata a signal or by using the unix socket. When Suricata is told to reload the rules these are the basic steps it takes:
+Suricata must be administratively directed to reload rules while it is running; there
+are two ways to cause this:
 
-* Load new config to update rule variables and values.
-* Load new rules
-* Construct new detection engine
-* Swap old and new detection engines
-* Make sure all threads are updated
-* Free old detection engine
+* Via a UNIX domain socket with the command ``reload-rules`` or ``ruleset-reload-nonblocking``
+* Upon receipt of the ``USR2`` signal
 
-Suricata will continue to process packets normally during this process. Keep in mind though, that the system should have enough memory for both detection engines.
+Suricata performs these steps while reloading rules:
 
-Signal::
+* The main Suricata configuration is reloaded to update rule variables and values; this
+  includes the rule related files ``classification.config``, ``reference.config`` and
+  ``theshold.config``
+* All rule files are reloaded with new rule variables applied
+* A new detection engine is created for the updated rules.
+* The previous and newly created detection engines are swapped.
+* Ensure all threads are updated
+* Free old detection engine and associated resources.
+
+Suricata will continue to process packets during the update process. Note that additional system
+memory is used during the reload process as a new detection engine and the reloaded rules are
+associated with it.
+
+Reload Triggers
+~~~~~~~~~~~~~~~
+
+Via process signal::
 
   kill -USR2 $(pidof suricata)
 
-There are two methods available when using the Unix socket.
+Via the UNIX domain socket.
 
 Blocking reload ::
 
   suricatasc -c reload-rules
 
-Non blocking reload ::
+Non-blocking reload ::
 
   suricatasc -c ruleset-reload-nonblocking
 
-It is also possible to get information about the last reload via dedicated commands. See :ref:`standard-unix-socket-commands` for more information.
+It is also possible to get information about the last reload via dedicated commands.
+See :ref:`standard-unix-socket-commands` for more information.
+
+Resources Reloaded
+~~~~~~~~~~~~~~~~~~
+
+There are two types of resources that are reloaded during a rule reload
+
+* Rule related configuration:
+  * Suricata's configuration file(s): ``suricata.yaml`` and any specified with the command-line
+    options ``--include <config-file.yaml>``
```

Comment:
Q: Is all of the configuration file `suricata.yaml` reloaded? There seem to be parts that'd be hard to reload while the engine is still up and running.

Reply:
I think we need an itemized list of all that is reloaded. However, I thought we already had that somewhere, but can't find it.

Reply:
The configuration file is reloaded -- however, not all values take effect.

The items in `vars` (address and port groups) take effect as rules are loaded.
Items in related files -- {threshold,reference,classification}.config take effect.

I'm not sure how to identify the other loaded values that take effect.